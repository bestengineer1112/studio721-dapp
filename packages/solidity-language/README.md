# solidity-language

A close-enough solidity AST
