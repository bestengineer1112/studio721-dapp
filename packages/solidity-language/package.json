{
  "name": "solidity-language",
  "version": "1.0.0",
  "main": "src/index.ts",
  "license": "UNLICENSED",
  "files": [
    "README.md"
  ]
}
