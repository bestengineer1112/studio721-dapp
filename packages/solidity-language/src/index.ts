export * as AST from './ast';
export * from './format';
export * from './builders';
