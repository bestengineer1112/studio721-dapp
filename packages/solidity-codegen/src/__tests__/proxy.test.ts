import {
  generateProxyApprovalFunction,
  generateProxyContracts,
} from '../generate/proxy';
import { contractDeclaration, formatContract, print } from 'solidity-language';

describe('proxy', () => {
  it('approval function', () => {
    const contract = contractDeclaration({
      name: 'Test',
      body: generateProxyApprovalFunction({
        usesDelegatedContract: false,
      }),
      extends: [],
    });

    expect(print(formatContract(contract))).toMatchSnapshot();
  });

  it('approval function with delegation', () => {
    const contract = contractDeclaration({
      name: 'Test',
      body: generateProxyApprovalFunction({
        usesDelegatedContract: true,
      }),
      extends: [],
    });

    expect(print(formatContract(contract))).toMatchSnapshot();
  });

  it('proxy contracts', () => {
    const contract = contractDeclaration({
      name: 'Test',
      body: generateProxyContracts(),
      extends: [],
    });

    expect(print(formatContract(contract))).toMatchSnapshot();
  });
});
