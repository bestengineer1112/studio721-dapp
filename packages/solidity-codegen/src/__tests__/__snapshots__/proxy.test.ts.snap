// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proxy approval function 1`] = `
"contract Test {
  /** PROXY REGISTRY **/

  address private immutable proxyRegistryAddress;

  function isApprovedForAll(address owner, address operator)
    override
    public
    view
    returns (bool)
  {
    ProxyRegistry proxyRegistry = ProxyRegistry(proxyRegistryAddress);

    if (address(proxyRegistry.proxies(owner)) == operator) {
      return true;
    }

    return super.isApprovedForAll(owner, operator);
  }
}"
`;

exports[`proxy approval function with delegation 1`] = `
"contract Test {
  /** PROXY REGISTRY **/

  address private immutable proxyRegistryAddress;

  function isApprovedForAll(address owner, address operator) public view
    returns (bool)
  {
    ProxyRegistry proxyRegistry = ProxyRegistry(proxyRegistryAddress);

    if (address(proxyRegistry.proxies(owner)) == operator) {
      return true;
    }

    return _isApprovedForAll(owner, operator);
  }
}"
`;

exports[`proxy proxy contracts 1`] = `
"contract Test {
  contract OwnableDelegateProxy {}

  contract ProxyRegistry {
    mapping(address => OwnableDelegateProxy) public proxies;
  }
}"
`;
